Consider Running SwiftLint in CI

In the process of investigating #82, I ran `swiftlint` on the project and found 10 violations of the  `implicit_return` rule:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'HyperbolicTests.swift' (1/26)
Linting 'XCTestCase+Surge.swift' (2/26)
Linting 'AuxiliaryTests.swift' (3/26)
Linting 'ExponentialTests.swift' (4/26)
Linting 'MatrixTests.swift' (5/26)
Linting 'TrigonometricTests.swift' (6/26)
Linting 'ArithmeticTests.swift' (7/26)
Linting 'ConvolutionTests.swift' (8/26)
Linting 'PowerTests.swift' (9/26)
Linting 'Contents.swift' (10/26)
Linting 'Package.swift' (11/26)
Linting 'Exponential.swift' (12/26)
Linting 'ArraySliceUnsafeMemory.swift' (13/26)
Linting 'Statistics.swift' (14/26)
Linting 'Arithmetic.swift' (15/26)
Linting 'Power.swift' (16/26)
Linting 'Pointers.swift' (17/26)
Linting 'Auxiliary.swift' (18/26)
Linting 'FFT.swift' (19/26)
Linting 'UnsafeMutableMemory.swift' (20/26)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:29:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:30:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:41:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:42:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:55:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:56:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:57:17: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:69:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:70:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:71:17: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
Linting 'Convolution.swift' (21/26)
Linting 'Trigonometric.swift' (22/26)
Linting 'Matrix.swift' (23/26)
Linting 'Hyperbolic.swift' (24/26)
Linting 'ArrayUnsafeMemory.swift' (25/26)
Linting 'UnsafeMemory.swift' (26/26)
Done linting! Found 10 violations, 0 serious in 26 files.
```

Currently, our CI is showing the build as passing, and project builds without warnings. Should we elevate linter diagnostics from its current advisory disposition to enforcement via a build phase or CI?

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,11 @@
 language: objective-c
 osx_image: xcode9.3
 
+install:
+  - brew update && brew install swiftlint
+
 script:
+  - swiftlint --strict
   - xcodebuild clean build test -workspace Surge.xcworkspace -scheme Surge-iOS -destination 'platform=iOS Simulator,name=iPhone 8,OS=11.1'
   - xcodebuild clean build test -workspace Surge.xcworkspace -scheme Surge-macOS -destination 'platform=OS X,arch=x86_64'
   - xcodebuild clean build test -workspace Surge.xcworkspace -scheme Surge-tvOS -destination 'platform=tvOS Simulator,name=Apple TV 4K,OS=11.1'

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: objective-c
-osx_image: xcode9.3
+osx_image: xcode9.4
 
 install:
-  - brew update && brew install swiftlint
+  - brew update && brew upgrade swiftlint
 
 script:
   - swiftlint --strict

